packages/engine/src/datastore/mod.rs

```diff
@@ -129,7 +129,7 @@ pub mod tests {
             .agent_pool()
             .batch(0)
             .unwrap()
-            .batch_id()
+            .segment_id()
```

Comment:
Everywhere else we seem to be doing `segment.memory.id`, is there a reason for this method?

Reply:
I simply missed this as I only ran `check` and `clippy`.